test/controllers/NiphlsNumberControllerSpec.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.SpecBase
+import controllers.actions.FakeAuthoriseAction
+import org.scalatest.matchers.should.Matchers.convertToAnyShouldWrapper
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import views.html.NiphlsNumberView
+import forms.NiphlsNumberFormProvider
+
+class NiphlsNumberControllerSpec extends SpecBase {
+
+  private val formProvider = new NiphlsNumberFormProvider()
+
+  private val niphlsNumberView = app.injector.instanceOf[NiphlsNumberView]
+
+  private val niphlsNumberController = new NiphlsNumberController(
+    stubMessagesControllerComponents(),
+    new FakeAuthoriseAction(defaultBodyParser),
+    niphlsNumberView,
+    formProvider
+  )
+
+  "NiphlsNumber Controller" - {
+
+    "must return OK and the correct view for a GET" in {
+
+      val result = niphlsNumberController.onPageLoad(fakeRequest)
+
+      status(result) mustEqual OK
+
+      contentAsString(result) mustEqual niphlsNumberView(formProvider())(fakeRequest, stubMessages()).toString
+
+    }
+
+    "must redirect on Submit when user enters valid NIPHL number" in {
```

Comment:
Can you the generator here:
`in forAll(generatorForNiphils.arbitrary) { niphl => …}`

Warning: sudo code

Reply:
Updated